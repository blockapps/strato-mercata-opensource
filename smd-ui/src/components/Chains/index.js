import React, { Component } from 'react';
import {
  fetchChains,
  fetchChainDetail,
  changeChainFilter,
  resetChainId,
  resetInitailLabel
} from './chains.actions';
import mixpanelWrapper from '../../lib/mixpanelWrapper';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Tour from '../Tour';
import CreateChain from '../CreateChain';
import Chain from '../Chain';
import './chains.css';
import HexText from '../HexText';
import { Button, Switch } from '@blueprintjs/core';
import ReactGA from 'react-ga4';

const tourSteps = [
  {
    title: 'View Chains',
    text: 'Scroll through all chains that you belong to',
    selector: '#chains',
    position: 'bottom',
    isFixed: true
  }
];

class Chains extends Component {
  constructor() {
    super()
    this.state = {
      selected: null,
      limit: 25,
      offset: 0,
      useChainIdSearch: false,
      chainId: "",
    }
  }

  componentDidMount() {
    this.props.fetchChains(this.state.limit, this.state.offset);
    mixpanelWrapper.track('chains_page_load');
    ReactGA.send({hitType: "pageview", page: "/shards", title: "Shards"});
  }

  updateLabelFilter(filter) {
    this.props.changeChainFilter(filter);
  };

  searchByChainId() {
    this.props.fetchChains(0, 0, this.state.chainId)
    
  }

  onUserClick(label, chainIds, index) {
    let uniqueKey = `${label}-${index}`
    if (chainIds.length && uniqueKey === this.state.selected) {
      this.props.resetChainId(label);
      this.setState({ selected: null });
    } else {
      mixpanelWrapper.track('chains_row_click');
      chainIds.forEach((chainId) => {
        this.props.fetchChainDetail(label, chainId);
      })
    }
  }

  onNextClick = () => {
    const { offset, limit } = this.state;
    const newOffset = offset + limit;
    this.setState({ offset: newOffset }, () => {
      this.props.fetchChains(this.state.limit, this.state.offset);
    });
  };

  onPrevClick = () => {
    const { offset, limit } = this.state;
    const newOffset = Math.max(0, offset - limit);
    this.setState({ offset: newOffset }, () => {
      this.props.fetchChains(this.state.limit, this.state.offset);
    });
  };

  render() {
    const labelIds = this.props.labelIds;
    const chains = this.props.chains;
    const filter = this.props.filter;
    const labels = Object.getOwnPropertyNames(chains);
    const rows = [];
    let selectedChains = [];

    labels.filter(label => {
      if (!filter) {
        return true;
      }
      return label
        .toLowerCase()
        .indexOf(filter) > -1
    })
      .forEach(function (label, index) {
        const chainIds = Object.getOwnPropertyNames(labelIds[label]);

        let labelClasseName = '';
        let uniqueKey = `${label}-${index}`
        if (this.state.selected === uniqueKey && chainIds.length > 0) {
          labelClasseName = ' selected';
          chainIds.map((chainid, key) =>
            selectedChains.push(<Chain label={label} id={chainid} chain={key} key={key} />)
          );
        }

        rows.push(
          <div className="smd-margin-8" key={label}>
            <div className="row">
              <div className={`pt-card pt-elevation-2 smd-pointer ${labelClasseName}`} key={index} onClick={(e) => {
                this.setState({ selected: uniqueKey });
                this.onUserClick(label, chainIds, index);
              }}>
                <HexText value={label} classes="large smd-pad-4 chain-width" />
              </div>
            </div>
          </div>
        );
      }.bind(this));

    const isPaginationDisplay = rows.length ? true : Boolean(this.state.offset);

    return (
      <div className="container-fluid pt-dark">
        <Tour
          name="chains"
          steps={tourSteps}
          finalStepSelector='#chains'
          nextPage='transactions' />

        <div className="row">
          <div className="col-sm-4 text-left">
            <h3>Shards</h3>
          </div>
          <div className="col-sm-8 text-right">
            <div className="pt-button-group">
              <CreateChain limit={this.state.limit} offset={this.state.offset} />
            </div>
          </div>
        </div>
        <div className="row" style={{display: 'flex', alignItems: 'center'}}>
          <div className="col-sm-4">
            <div className="pt-input-group pt-dark pt-large">
              <span className="pt-icon pt-icon-search"></span>
              <input
                className="pt-input"
                type="search"
                placeholder={`Search Shards (${this.state.useChainIdSearch ? "Shard ID" : "Label"})`}
                onChange={e => {
                  this.state.useChainIdSearch ?
                    this.setState({chainId : e.target.value}) :
                    this.updateLabelFilter(e.target.value.toLowerCase())
                }}
                dir="auto" />
            </div>
          </div >
          <div className="col-sm-2">
            {
              this.state.useChainIdSearch ?
              <button 
                className='pt-button pt-icon-search' 
                onClick={() => this.searchByChainId()}
                >
                  Search
                </button> : undefined
            }
          </div>
          <div className='col-sm-2'>
            <Switch
              checked={this.state.useChainIdSearch}
              onChange={() => {
                this.setState({useChainIdSearch: !this.state.useChainIdSearch}, () => {
                  if (!this.state.useChainIdSearch) {
                    this.props.fetchChains(this.state.limit, this.state.offset)
                    this.setState({chainId : ""})
                  }

                })}}
              label="Search by Shard ID"
              />
          </div>
          
        </div>
        <div className="container-fluid pt-dark">
          <div className="row">
            <div className="col-sm-4 main-div">
              <div className="accounts-margin-top">
                {!rows.length && !this.props.isLoading &&
                  <table>
                    <tbody>
                      <tr>
                        <td colSpan={3}>No Shards</td>
                      </tr>
                    </tbody>
                  </table>}
                  {
                    this.props.isLoading ? <table>
                    <tbody>
                      <tr>
                        <td colSpan={3}>Fetching....</td>
                      </tr>
                    </tbody>
                  </table>
                  : rows
                  }
              </div>
            </div>
            <div className="col-sm-8 account-details">
              <div>
                {selectedChains}
              </div>
            </div>
          </div>
          {isPaginationDisplay &&
            <div className="row">
              <div className="col-sm-1 smd-pad-16 text-left">
                <Button
                  onClick={this.onPrevClick}
                  className="pt-icon-arrow-left"
                  text="Previous"
                  disabled={!(this.state.offset > 0)}
                />
              </div>
              <div className="col-sm-2 text-center" style={{ marginTop: '22px' }}>
                {`Rows ${this.state.offset + 1}-${this.state.offset + Math.min(rows.length, this.state.limit)}`}
              </div>
              <div className="col-sm-1 smd-pad-16 text-right">
                <Button
                  onClick={this.onNextClick}
                  className="pt-icon-arrow-right"
                  text="Next"
                  disabled={rows.length < this.state.limit}
                />
              </div>
            </div>}
        </div>
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {
    filter: state.chains.filter,
    chains: state.chains.chains,
    labelIds: state.chains.labelIds,
    initialLabel: state.chains.initialLabel,
    isLoading: state.chains.isLoading,
  };
}

export default withRouter(
  connect(mapStateToProps,
    {
      fetchChains,
      fetchChainDetail,
      resetChainId,
      changeChainFilter,
      resetInitailLabel
    }
  )(Chains));