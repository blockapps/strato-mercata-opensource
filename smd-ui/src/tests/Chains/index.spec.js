import React from 'react';
import Chains, { mapStateToProps } from '../../components/Chains';
import { chain } from './chainsMock';

describe('Chains: index', () => {

  describe('render component', () => {

    test('with empty values', () => {
      const props = {
        labelIds: [],
        chains: [],
        filter: null,
        isLoading: false,
        changeChainFilter: jest.fn(),
        resetChainId: jest.fn(),
        fetchChains: jest.fn(),
        fetchChainDetail: jest.fn()
      };

      const wrapper = shallow(
        <Chains.WrappedComponent {...props} />
      );

      expect(wrapper.debug()).toMatchSnapshot();
      expect(props.fetchChains).toHaveBeenCalled();
      expect(props.fetchChains).toHaveBeenCalledTimes(1);
    });

    test('with values', () => {
      const props = {
        labelIds: chain,
        chains: chain,
        filter: null,
        isLoading: true,
        changeChainFilter: jest.fn(),
        resetChainId: jest.fn(),
        fetchChains: jest.fn(),
        fetchChainDetail: jest.fn()
      };

      const wrapper = shallow(
        <Chains.WrappedComponent {...props} />
      );

      expect(wrapper.debug()).toMatchSnapshot();
      expect(props.fetchChains).toHaveBeenCalled();
      expect(props.fetchChains).toHaveBeenCalledTimes(1);
    });

  });

  test('filter chain', () => {
    const props = {
      labelIds: chain,
      chains: chain,
      filter: null,
      isLoading: false,
      changeChainFilter: jest.fn(),
      resetChainId: jest.fn(),
      fetchChains: jest.fn(),
      fetchChainDetail: jest.fn()
    };

    const wrapper = shallow(
      <Chains.WrappedComponent {...props} />
    );

    wrapper.find('input').simulate('change', { target: { value: 'airline cartel 1' } });
    expect(props.changeChainFilter).toHaveBeenCalled();
    expect(props.changeChainFilter).toHaveBeenCalledTimes(1);
  });

  test('show/hide chain detail', () => {
    const props = {
      labelIds: chain,
      chains: chain,
      filter: null,
      isLoading: false,
      changeChainFilter: jest.fn(),
      resetChainId: jest.fn(),
      fetchChains: jest.fn(),
      fetchChainDetail: jest.fn()
    };

    const wrapper = shallow(
      <Chains.WrappedComponent {...props} />
    );

    wrapper.find('div').at(19).simulate('click');
    expect(props.fetchChainDetail).toHaveBeenCalled();
    expect(props.fetchChainDetail).toHaveBeenCalledTimes(1);
    wrapper.find('div').at(19).simulate('click');
    expect(props.resetChainId).toHaveBeenCalled();
    expect(props.resetChainId).toHaveBeenCalledTimes(1);
  });

  test('filter chain', () => {
    const props = {
      labelIds: chain,
      chains: chain,
      filter: 'airline cartel',
      isLoading: false,
      changeChainFilter: jest.fn(),
      resetChainId: jest.fn(),
      fetchChains: jest.fn(),
      fetchChainDetail: jest.fn()
    };

    const wrapper = shallow(
      <Chains.WrappedComponent {...props} />
    );

    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('mapStateToProps with default state', () => {
    const state = {
      chains: {
        filter: 'airline',
        chains: chain,
        labelIds: chain,
        initialLabel: null
      }
    }

    expect(mapStateToProps(state)).toMatchSnapshot();
  });

});