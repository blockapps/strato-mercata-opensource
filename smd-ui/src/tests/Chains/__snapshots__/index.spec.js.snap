// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chains: index filter chain 1`] = `
"<div className=\\"container-fluid pt-dark\\">
  <withRouter(Connect(Tour)) name=\\"chains\\" steps={{...}} finalStepSelector=\\"#chains\\" nextPage=\\"transactions\\" />
  <div className=\\"row\\">
    <div className=\\"col-sm-4 text-left\\">
      <h3>
        Shards
      </h3>
    </div>
    <div className=\\"col-sm-8 text-right\\">
      <div className=\\"pt-button-group\\">
        <withRouter(Connect(ReduxForm)) limit={25} offset={0} />
      </div>
    </div>
  </div>
  <div className=\\"row\\" style={{...}}>
    <div className=\\"col-sm-4\\">
      <div className=\\"pt-input-group pt-dark pt-large\\">
        <span className=\\"pt-icon pt-icon-search\\" />
        <input className=\\"pt-input\\" type=\\"search\\" placeholder=\\"Search Shards (Label)\\" onChange={[Function: onChange]} dir=\\"auto\\" />
      </div>
    </div>
    <div className=\\"col-sm-2\\" />
    <div className=\\"col-sm-2\\">
      <Blueprint.Switch checked={false} onChange={[Function: onChange]} label=\\"Search by Shard ID\\" />
    </div>
  </div>
  <div className=\\"container-fluid pt-dark\\">
    <div className=\\"row\\">
      <div className=\\"col-sm-4 main-div\\">
        <div className=\\"accounts-margin-top\\">
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 9\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 1\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 2\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 3\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 4\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 5\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 6\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 7\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 10\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 11\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 12\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 13\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 14\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 89\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 200\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 822\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 52\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 900\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 1000\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 9000\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"airline cartel 2000\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className=\\"col-sm-8 account-details\\">
        <div />
      </div>
    </div>
    <div className=\\"row\\">
      <div className=\\"col-sm-1 smd-pad-16 text-left\\">
        <Blueprint.Button onClick={[Function]} className=\\"pt-icon-arrow-left\\" text=\\"Previous\\" disabled={true} />
      </div>
      <div className=\\"col-sm-2 text-center\\" style={{...}}>
        Rows 1-21
      </div>
      <div className=\\"col-sm-1 smd-pad-16 text-right\\">
        <Blueprint.Button onClick={[Function]} className=\\"pt-icon-arrow-right\\" text=\\"Next\\" disabled={true} />
      </div>
    </div>
  </div>
</div>"
`;

exports[`Chains: index mapStateToProps with default state 1`] = `
Object {
  "chains": Object {
    "airline cartel 1": Object {
      "79f69ec8f4bdb4a0c43e8970e4f2a9701db43b8b7b046023fe2a874ddb32acae": Object {},
    },
    "airline cartel 10": Object {
      "b3e7a42333819dcb5bf800b7cc74f59eb9861d4ccf8d8138eb8f3d4d9308e5a7": Object {},
    },
    "airline cartel 1000": Object {
      "63e0a0b02a9216d980dde66215d2c2e3b598f58c421d0046c100b4e0a571f40c": Object {},
    },
    "airline cartel 11": Object {
      "1f0744737a6b9e2a0688cc7bb58d3696fe777fe53659613ba164c82a9f210fb4": Object {},
    },
    "airline cartel 12": Object {
      "2206661f918ca9f1fad4578b212f3e5abff7ad714e024297a1ba70fdcf52be67": Object {},
    },
    "airline cartel 13": Object {
      "b5f6dc91e333b20140b007c55f28027be28bb8b54e744fbed609668293376efc": Object {},
      "f8d3a497da2c7325eab38ed8a48c99e463c4417bf75e1694327f9633fffcaa10": Object {},
      "fb92498236dfeb519ff6d36a7bc37afd4eab34e05f5d40eabc3539621c09217b": Object {},
    },
    "airline cartel 14": Object {
      "2f19b069f2700ffd7217025efb1bcf47603a6a137a07b0c28bbedc47f0485acf": Object {},
    },
    "airline cartel 2": Object {
      "e5ea4617812076bac15e12e4f70f78076f9f371ac744022905be97e57ca05bf2": Object {},
    },
    "airline cartel 200": Object {
      "2b1ead18005e051e3fb68ce0f00075f4fa289144b1d490c0a04c7c1a90437021": Object {},
    },
    "airline cartel 2000": Object {
      "5b89dd643d729fe9d433f5243acefc596036aacd428499e6994832e5370d4ef9": Object {},
    },
    "airline cartel 3": Object {
      "bd3d6838e2bdcb5b1cf1093873d90e3192aa8bcffe05d1b2d6522f09b4dafef8": Object {},
    },
    "airline cartel 4": Object {
      "bbee98fb39586c13884998ccdeed659abf9b44c2c3eede252f57c83d4245e66c": Object {},
    },
    "airline cartel 5": Object {
      "092cec7cd999b8b4827e0e384d2d8636e8484b6e2d8f824d4727663cd1ba330d": Object {},
    },
    "airline cartel 52": Object {
      "4a8512095130296c0388d3c8277228890ee5b10ee83383ce0be1dd24e3537b7d": Object {},
    },
    "airline cartel 6": Object {
      "6a5abc53876eaae27279b194335526b52c3980e19dff5ab0de384deed1a90950": Object {},
    },
    "airline cartel 7": Object {
      "0d705c5d05ae4cc780655d92939cdb31594ffa8b29c5a26a6ae0bb6b2a66b2c4": Object {},
    },
    "airline cartel 822": Object {
      "6913f2c6faa6e47746c0dae953536116941bdf8473ae2452a2ab6ae01588b2a6": Object {},
    },
    "airline cartel 89": Object {
      "54d04b91157066f2bed14954c7c635b2cd82dc754db4195c61e90e208d9bd5d5": Object {},
    },
    "airline cartel 9": Object {
      "75dc24995abf63fe7d637b4879353a41593ef05c37ee6d11704bb97403306a86": Object {},
      "86d43e5b7a57711ecf734b79fb998c119d525afa2832ee726d6791cdac1864c6": Object {},
      "8768cb0769067d78b147974d627884b4de0737d60ea2f4e978663a62946830a5": Object {},
      "8eeefc708630002bb349ce1e1f914ac3adf8efc385106a1ef04b90c3d9b7ee60": Object {},
      "90d35b73a7ca7dd4fe4ccfd425bf404f92eb3091e502643bc62b936e331ad08b": Object {},
      "936986d54af89e2fa3c4888563856940d8f80cd5c1d01c386bd98207d29ae1e9": Object {},
      "ab1696d5f6a36836f0bd23d96918cce4f78b3667518d70e721914f391142a5e1": Object {},
    },
    "airline cartel 900": Object {
      "c861f9caed98de482888784ba0384e7a23a086d1387702ac3981b81d8df30632": Object {},
    },
    "airline cartel 9000": Object {
      "8d30f74bfe22245ba513ee0d09bbcc0d26f82513fb35bdc9b5862d5c5c88dcf6": Object {},
    },
  },
  "filter": "airline",
  "initialLabel": null,
  "isLoading": undefined,
  "labelIds": Object {
    "airline cartel 1": Object {
      "79f69ec8f4bdb4a0c43e8970e4f2a9701db43b8b7b046023fe2a874ddb32acae": Object {},
    },
    "airline cartel 10": Object {
      "b3e7a42333819dcb5bf800b7cc74f59eb9861d4ccf8d8138eb8f3d4d9308e5a7": Object {},
    },
    "airline cartel 1000": Object {
      "63e0a0b02a9216d980dde66215d2c2e3b598f58c421d0046c100b4e0a571f40c": Object {},
    },
    "airline cartel 11": Object {
      "1f0744737a6b9e2a0688cc7bb58d3696fe777fe53659613ba164c82a9f210fb4": Object {},
    },
    "airline cartel 12": Object {
      "2206661f918ca9f1fad4578b212f3e5abff7ad714e024297a1ba70fdcf52be67": Object {},
    },
    "airline cartel 13": Object {
      "b5f6dc91e333b20140b007c55f28027be28bb8b54e744fbed609668293376efc": Object {},
      "f8d3a497da2c7325eab38ed8a48c99e463c4417bf75e1694327f9633fffcaa10": Object {},
      "fb92498236dfeb519ff6d36a7bc37afd4eab34e05f5d40eabc3539621c09217b": Object {},
    },
    "airline cartel 14": Object {
      "2f19b069f2700ffd7217025efb1bcf47603a6a137a07b0c28bbedc47f0485acf": Object {},
    },
    "airline cartel 2": Object {
      "e5ea4617812076bac15e12e4f70f78076f9f371ac744022905be97e57ca05bf2": Object {},
    },
    "airline cartel 200": Object {
      "2b1ead18005e051e3fb68ce0f00075f4fa289144b1d490c0a04c7c1a90437021": Object {},
    },
    "airline cartel 2000": Object {
      "5b89dd643d729fe9d433f5243acefc596036aacd428499e6994832e5370d4ef9": Object {},
    },
    "airline cartel 3": Object {
      "bd3d6838e2bdcb5b1cf1093873d90e3192aa8bcffe05d1b2d6522f09b4dafef8": Object {},
    },
    "airline cartel 4": Object {
      "bbee98fb39586c13884998ccdeed659abf9b44c2c3eede252f57c83d4245e66c": Object {},
    },
    "airline cartel 5": Object {
      "092cec7cd999b8b4827e0e384d2d8636e8484b6e2d8f824d4727663cd1ba330d": Object {},
    },
    "airline cartel 52": Object {
      "4a8512095130296c0388d3c8277228890ee5b10ee83383ce0be1dd24e3537b7d": Object {},
    },
    "airline cartel 6": Object {
      "6a5abc53876eaae27279b194335526b52c3980e19dff5ab0de384deed1a90950": Object {},
    },
    "airline cartel 7": Object {
      "0d705c5d05ae4cc780655d92939cdb31594ffa8b29c5a26a6ae0bb6b2a66b2c4": Object {},
    },
    "airline cartel 822": Object {
      "6913f2c6faa6e47746c0dae953536116941bdf8473ae2452a2ab6ae01588b2a6": Object {},
    },
    "airline cartel 89": Object {
      "54d04b91157066f2bed14954c7c635b2cd82dc754db4195c61e90e208d9bd5d5": Object {},
    },
    "airline cartel 9": Object {
      "75dc24995abf63fe7d637b4879353a41593ef05c37ee6d11704bb97403306a86": Object {},
      "86d43e5b7a57711ecf734b79fb998c119d525afa2832ee726d6791cdac1864c6": Object {},
      "8768cb0769067d78b147974d627884b4de0737d60ea2f4e978663a62946830a5": Object {},
      "8eeefc708630002bb349ce1e1f914ac3adf8efc385106a1ef04b90c3d9b7ee60": Object {},
      "90d35b73a7ca7dd4fe4ccfd425bf404f92eb3091e502643bc62b936e331ad08b": Object {},
      "936986d54af89e2fa3c4888563856940d8f80cd5c1d01c386bd98207d29ae1e9": Object {},
      "ab1696d5f6a36836f0bd23d96918cce4f78b3667518d70e721914f391142a5e1": Object {},
    },
    "airline cartel 900": Object {
      "c861f9caed98de482888784ba0384e7a23a086d1387702ac3981b81d8df30632": Object {},
    },
    "airline cartel 9000": Object {
      "8d30f74bfe22245ba513ee0d09bbcc0d26f82513fb35bdc9b5862d5c5c88dcf6": Object {},
    },
  },
}
`;

exports[`Chains: index render component with empty values 1`] = `
"<div className=\\"container-fluid pt-dark\\">
  <withRouter(Connect(Tour)) name=\\"chains\\" steps={{...}} finalStepSelector=\\"#chains\\" nextPage=\\"transactions\\" />
  <div className=\\"row\\">
    <div className=\\"col-sm-4 text-left\\">
      <h3>
        Shards
      </h3>
    </div>
    <div className=\\"col-sm-8 text-right\\">
      <div className=\\"pt-button-group\\">
        <withRouter(Connect(ReduxForm)) limit={25} offset={0} />
      </div>
    </div>
  </div>
  <div className=\\"row\\" style={{...}}>
    <div className=\\"col-sm-4\\">
      <div className=\\"pt-input-group pt-dark pt-large\\">
        <span className=\\"pt-icon pt-icon-search\\" />
        <input className=\\"pt-input\\" type=\\"search\\" placeholder=\\"Search Shards (Label)\\" onChange={[Function: onChange]} dir=\\"auto\\" />
      </div>
    </div>
    <div className=\\"col-sm-2\\" />
    <div className=\\"col-sm-2\\">
      <Blueprint.Switch checked={false} onChange={[Function: onChange]} label=\\"Search by Shard ID\\" />
    </div>
  </div>
  <div className=\\"container-fluid pt-dark\\">
    <div className=\\"row\\">
      <div className=\\"col-sm-4 main-div\\">
        <div className=\\"accounts-margin-top\\">
          <div className=\\"smd-margin-8\\">
            <div className=\\"row\\">
              <div className=\\"pt-card pt-elevation-2 smd-pointer \\" onClick={[Function: onClick]}>
                <HexText value=\\"length\\" classes=\\"large smd-pad-4 chain-width\\" />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className=\\"col-sm-8 account-details\\">
        <div />
      </div>
    </div>
    <div className=\\"row\\">
      <div className=\\"col-sm-1 smd-pad-16 text-left\\">
        <Blueprint.Button onClick={[Function]} className=\\"pt-icon-arrow-left\\" text=\\"Previous\\" disabled={true} />
      </div>
      <div className=\\"col-sm-2 text-center\\" style={{...}}>
        Rows 1-1
      </div>
      <div className=\\"col-sm-1 smd-pad-16 text-right\\">
        <Blueprint.Button onClick={[Function]} className=\\"pt-icon-arrow-right\\" text=\\"Next\\" disabled={true} />
      </div>
    </div>
  </div>
</div>"
`;

exports[`Chains: index render component with values 1`] = `
"<div className=\\"container-fluid pt-dark\\">
  <withRouter(Connect(Tour)) name=\\"chains\\" steps={{...}} finalStepSelector=\\"#chains\\" nextPage=\\"transactions\\" />
  <div className=\\"row\\">
    <div className=\\"col-sm-4 text-left\\">
      <h3>
        Shards
      </h3>
    </div>
    <div className=\\"col-sm-8 text-right\\">
      <div className=\\"pt-button-group\\">
        <withRouter(Connect(ReduxForm)) limit={25} offset={0} />
      </div>
    </div>
  </div>
  <div className=\\"row\\" style={{...}}>
    <div className=\\"col-sm-4\\">
      <div className=\\"pt-input-group pt-dark pt-large\\">
        <span className=\\"pt-icon pt-icon-search\\" />
        <input className=\\"pt-input\\" type=\\"search\\" placeholder=\\"Search Shards (Label)\\" onChange={[Function: onChange]} dir=\\"auto\\" />
      </div>
    </div>
    <div className=\\"col-sm-2\\" />
    <div className=\\"col-sm-2\\">
      <Blueprint.Switch checked={false} onChange={[Function: onChange]} label=\\"Search by Shard ID\\" />
    </div>
  </div>
  <div className=\\"container-fluid pt-dark\\">
    <div className=\\"row\\">
      <div className=\\"col-sm-4 main-div\\">
        <div className=\\"accounts-margin-top\\">
          <table>
            <tbody>
              <tr>
                <td colSpan={3}>
                  Fetching....
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div className=\\"col-sm-8 account-details\\">
        <div />
      </div>
    </div>
    <div className=\\"row\\">
      <div className=\\"col-sm-1 smd-pad-16 text-left\\">
        <Blueprint.Button onClick={[Function]} className=\\"pt-icon-arrow-left\\" text=\\"Previous\\" disabled={true} />
      </div>
      <div className=\\"col-sm-2 text-center\\" style={{...}}>
        Rows 1-21
      </div>
      <div className=\\"col-sm-1 smd-pad-16 text-right\\">
        <Blueprint.Button onClick={[Function]} className=\\"pt-icon-arrow-right\\" text=\\"Next\\" disabled={true} />
      </div>
    </div>
  </div>
</div>"
`;
